mods/core/automount.lua

```diff
@@ -1,12 +1,96 @@
-automount = function(player)
-    local fields = {
-        connection_string = os.getenv("INFERNO_ADDRESS") ~= "" and os.getenv("INFERNO_ADDRESS") or
-            core_conf:get("inferno_address"),
-        connect = true
-    }
-    connect(player, fields)
+automount = function()
+    -- get string in form of tcp!host!port from ENV or mod.conf
+    local attach_string = os.getenv("INFERNO_ADDRESS") ~= "" and os.getenv("INFERNO_ADDRESS") or
+                              core_conf:get("inferno_address")
+
+    -- establish 9p attachment
+    local conn = connections[attach_string]
+    if not conn then
+        conn = connection(attach_string)
+        if not conn:attach() then
+            error("Failed connecting to the inferno os")
+        end
+    elseif conn:is_alive() then
+        print("Already attached. Connection is alive")
+    elseif conn.tcp then
+        print("Connection is not alive. Reconnecting")
+        conn:reattach()
+    else
+        if not conn:attach() then
+            error("Failed connecting to the inferno os")
+        end
+    end
+
+    -- check for presence of cmdchan
+    local cmdchan_path = tostring(core_conf:get("cmdchan_path"))
+    local root_cmdchan = cmdchan(conn, cmdchan_path)
+    if not root_cmdchan:is_present() then
+        error("cmdchan at path " .. cmdchan_path .. " is not available")
+    else
+        print("cmdchan is available")
+    end
+
+    -- mount registry
+    print(root_cmdchan:execute("mkdir -p /n/9mine /mnt/registry"))
+    print(root_cmdchan:execute("mount -A tcp!registry.dev.metacoma.io!30100 /mnt/registry"))
+
+    -- get and mount user management service
+    local user_management = root_cmdchan:execute("ndb/regquery -n description 'user management'"):gsub("\n", "")
+    while not user_management:match(".*!.*!.*") do
+        os.execute("sleep 2")
```

Comment:
all clients will stuck for 2 seconds, use minetest.after instead of sleep

Reply:
Fixed. Minetest server will look for user management every 3 second with minetest.after